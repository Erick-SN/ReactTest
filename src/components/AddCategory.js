import { useState } from 'react';
const AddCategory = () => {
  const [input, setInput] = useState();
  const handleInputChange = (e) => {
    setInput(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
  };
  return (
    <>
      <form onSubmit={handleSubmit}>
        <input type="text" value={input} onChange={handleInputChange} />
      </form>
    </>
  );
};

export default AddCategory;
