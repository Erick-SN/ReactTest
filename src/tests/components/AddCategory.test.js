import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas al componente <AddCategory/>', () => {
  let wrapper;
  const setCategories = jest.fn();
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test('renderizado del componente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Validando evento del input', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';
    input.simulate('change', { target: { value } });
  });
  test('no debe postear la informacion', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
  test('enviar la informacion', () => {
    const value = 'Goku';
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
