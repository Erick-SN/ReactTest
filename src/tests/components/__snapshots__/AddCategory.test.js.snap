// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <AddCategory/> renderizado del componente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
